compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -313,7 +313,10 @@ private boolean isDistinctSymbol(BSymbol symbol, BSymbol foundSym) {
 
     private boolean hasSameOwner(BSymbol symbol, BSymbol foundSym) {
         // check whether the given symbol owner is same as found symbol's owner
-        if (foundSym.owner == symbol.owner) {
+        if (foundSym.owner == symbol.owner ||
+                (foundSym.owner.tag & SymTag.PACKAGE) == SymTag.PACKAGE &&
+                (symbol.owner.tag & SymTag.PACKAGE) == SymTag.PACKAGE &&
+                foundSym.pkgID.equals(symbol.pkgID)) {
```

Comment:
Isn't this issue due to/related to https://github.com/ballerina-platform/ballerina-lang/issues/26284, https://github.com/ballerina-platform/ballerina-lang/issues/30688. etc. 

```ballerina
public enum Foo {
    foo, 
    bar
}

type Foo int;
```

The owner of the symbol for `type Foo int` is what is incorrect here (lang.annotations), right? I think we need to fix the type aliasing issues before/instead of introducing these changes here.

@pubudu91, thoughts please?

Reply:
This check is ok I guess. But wondering how this fixed the issue though. Like Maryam said, in the type def's symbol the owner would be the `lang.annotations` module. 🤔

Reply:
We do have a symbol for `Foo` in `type Foo int;` here and the `pkgID` is the current package's ID. It's just that the owner of that is the lang.annotations module.

We shouldn't need this check though, right? I think we should be able to rely on the owner. IINM, there shouldn't be scenarios where two symbols' package IDs are the same but the owner package symbols are different?

Reply:
Ah I see. Yeah, with the fix for type aliases, ideally shouldn't need this. IIRC, we don't create package symbols like we do for other kinds of symbols. But for now, don't see any harm in this though.